app/adapter/graphql/resolver/url_test.go

```diff
@@ -0,0 +1,38 @@
+package resolver
+
+import (
+	"github.com/byliuyang/app/mdtest"
+	"short/app/adapter/graphql/scalar"
+	"short/app/entity"
+	"testing"
+	"time"
+)
+
+func TestURL_Alias(t *testing.T) {
+	urlResolver := URL{url: entity.URL{Alias: "TestAlias"}}
+
+	expected := urlResolver.url.Alias
+	got := *urlResolver.Alias()
+	mdtest.Equal(t, got, expected, "*urlTest.Alias() = %v; want %v", expected, got)
+}
+
+func TestURL_OriginalURL(t *testing.T) {
+	urlResolver := URL{url: entity.URL{OriginalURL: "TestOriginalUrl"}}
+
+	expected := urlResolver.url.OriginalURL
+	got := *urlResolver.OriginalURL()
+	mdtest.Equal(t, got, expected, "*urlResolver.OriginalURL() = %v; want %v", expected, got)
+}
+
+func TestURL_ExpireAt(t *testing.T) {
+	timeAfter := time.Now().Add(5 * time.Second)
+	urlResolver := URL{url: entity.URL{ExpireAt: &timeAfter}}
+
+	expected := scalar.Time{Time: *urlResolver.url.ExpireAt}.Time
+	got := urlResolver.ExpireAt().Time
+	mdtest.Equal(t, got, expected, "*urlResolver.OriginalURL() = %v; want %v", expected, got)
+
+	urlResolver.url.ExpireAt = nil
```

Comment:
It seems that we are testing for two groups of inputs. I wonder whether we can refactor this test to do table testing instead. 

https://golang.org/doc/code.html#Testing

https://cs.time4hacks.com/github.com/byliuyang/short@23d127c373c990d8fa1210758a20c5b2c9b23f6d/-/blob/backend/app/usecase/auth/authenticator_test.go

Reply:
I'll take a look now.
What's in the 2nd link? It requires github authorization.

Reply:
@dedifferentiator It is called Sourcegraph, the code search tool we use.

Reply:
Who are we? :)

Reply:
@dedifferentiator Both of us now. We use code search at Google. Uber & Yelp use Sourcegraph instead :)